File: VB/reporting-provide-custom-report-gallery-storage-in-report-designer/XpoReportGalleryStorage/ReportGalleryTable.vb
Imports DevExpress.Xpo
Imports DevExpress.XtraReports.ReportGallery

Namespace XpoReportGalleryStorage

    Public Class ReportGalleryTable
        Inherits XPLiteObject

        Private sIDField As String

        Private displayNameField As String

        Private hashField As Integer

        Private contentHashField As Integer

        Private galleryItemType As GalleryItemType

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub

        <Key>
        Public Property ID As Integer

        Public Property SID As String
            Get
                Return sIDField
            End Get

            Set(ByVal value As String)
                SetPropertyValue("SID", sIDField, value)
            End Set
        End Property

        Public Property ItemKind As String

        Public Property DisplayName As String
            Get
                Return displayNameField
            End Get

            Set(ByVal value As String)
                SetPropertyValue("DisplayName", displayNameField, value)
            End Set
        End Property

        Public Property Content As Byte()

        Public Property Hash As Integer
            Get
                Return hashField
            End Get

            Set(ByVal value As Integer)
                SetPropertyValue("Hash", hashField, value)
            End Set
        End Property

        Public Property ContentHash As Integer
            Get
                Return contentHashField
            End Get

            Set(ByVal value As Integer)
                SetPropertyValue("ContentHash", contentHashField, value)
            End Set
        End Property

        Public Property ItemType As GalleryItemType
            Get
                Return galleryItemType
            End Get

            Set(ByVal value As GalleryItemType)
                SetPropertyValue("ItemType", galleryItemType, value)
            End Set
        End Property
    End Class
End Namespace

File: CS/reporting-provide-custom-report-gallery-storage-in-report-designer/XpoReportGalleryStorage/ReportGalleryTable.cs
using DevExpress.Xpo;
using DevExpress.XtraReports.ReportGallery;

namespace XpoReportGalleryStorage {
    public class ReportGalleryTable : XPLiteObject {
        string sID;
        string displayName;
        int hash;
        int contentHash;
        GalleryItemType galleryItemType;

        public ReportGalleryTable(Session session) : base(session) {
        }

        [Key]
        public int ID { get; set; }

        public string SID {
            get { return sID; }
            set { SetPropertyValue("SID", ref sID, value); }
        }

        public string ItemKind { get; set; }

        public string DisplayName {
            get { return displayName; }
            set { SetPropertyValue("DisplayName", ref displayName, value); }
        }

        public byte[] Content { get; set; }

        public int Hash {
            get { return hash; }
            set { SetPropertyValue("Hash", ref hash, value); }
        }

        public int ContentHash {
            get { return contentHash; }
            set { SetPropertyValue("ContentHash", ref contentHash, value); }
        }

        public GalleryItemType ItemType {
            get { return galleryItemType; }
            set { SetPropertyValue("ItemType", ref galleryItemType, value); }
        }

    }
}
